### ISSUE TITLE
Large result sets cause Colab to crash

### ISSUE THREAD

**Issue by duncandewhurst** [OpenDataServices/noteql#7]
[Example](https://colab.research.google.com/drive/18o2_v9obXOTIquqVLSDcrwvFTgewzF5l#scrollTo=BSAOvoa__PjP&line=5&uniqifier=1).

I can run this query using ipython-sql, although it does take a long time to return the results.

The context for returning so many results is that I was trying to get a random sample from a whole dataset by using `ORDER BY random()` in a SQL query, but the query was slow, so instead I am trying to return all the results and do the sampling in Pandas instead.

**Comment by kindly** [comment]
That is interesting as I am using the pandas read_sql command which I thought would be the best.

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_sql.html

however maybe I need to set the chunksize to a higher value which I think will help for larger datasets.

**Comment by duncandewhurst** [comment]
Ah, actually the issue was that I tried both methods in the same notebook, so noteql had less RAM available because ipython-sql had already used some.

I restarted the runtime between tests and the query completed successfully. Both libraries used similar amounts of RAM.

In both cases, returning the results took a lot longer than running the query:

- query execution time according to EXPLAIN ANALYZE: 35s
- cell execution time with noteql: 127s
- cell execution time with ipython-sql: 106s

Edit: Actual cell execution times varied in my earlier testing, so the above results may not be representative as they are from only one execution for each library.
